activerecord/lib/active_record/encryption/encrypted_attribute_type.rb

```diff
@@ -136,7 +136,7 @@ def serialize_with_current(value)
         def encrypt_as_text(value)
           with_context do
             encryptor.encrypt(value, **encryption_options).tap do |encrypted|
-              if !cast_type.binary? && encrypted.encoding == Encoding::BINARY
+              if encryptor.binary? && !cast_type.binary? && encrypted.encoding == Encoding::BINARY
```

Comment:
```suggestion
              if encryptor.binary? && !cast_type.binary?
```

If the encryptor reports that it produces binary data that should be enough I think, we don't also need to check the encrypted value is binary.

Reply:
Addressed in [these changes](https://github.com/rails/rails/compare/1d71c046fa9e318319855c46657e7eb0d1bbc77b..cf3c3a4e36d149a48b623591b7babf0bf107be2e).